compiler-core/src/build.rs

```diff
@@ -185,6 +186,7 @@ impl Module {
 #[derive(Debug, Clone, PartialEq)]
 pub enum Located<'a> {
     Expression(&'a TypedExpr),
+    Pattern(&'a TypedPattern, Type),
```

Comment:
What's the best approach for the lifetime of Type? Should it be a reference?
I ran into issues here...